kb/deadline-exceeded-troubleshooting.md

```diff
@@ -0,0 +1,104 @@
+Network interruptions, short timeouts, server overload, and Query errors can cause this error.
+
+The following sections will discuss the nature of this error and how to troubleshoot it.
+
+#### Check your logs
+
+:::note
+
+Cloud users cannot access some of the logs needed to diagnose the source of the error.
+If you're using Temporal Cloud, consider filing a [service ticket](https://support.temporal.io/) when this error occurs.
+
+:::
+
+```
+{“level”:“error”,“ts”:“2022-03-21T19:32:42.312Z”,“msg”:“unavailable error”,“service”:“frontend”,“error”:“unable to get temporal-sys-add-search-attributes-workflow workflow state: context deadline exceeded”,“logging-call-at”:“adminHandler.go:1163”,“stacktrace”:“go.temporal.io/server/common/log.
+```
+
+Service logs can show which parts of the Cluster aren't working.
+
+Verify that the frontend is connected by opening the Web UI in your browser.
+
+Check that the following are up and running with `tctl cluster` commands:
+
+- [History Service](/concepts/what-is-a-history-service)
+- [Persistence](/clusters#persistence) database
+- [Advanced Visibility](/concepts/what-is-advanced-visibility) database
+
+Logs can also be used to find [Query](/concepts/what-is-a-query) errors.
+
+#### Check your Cluster metrics
+
+Look for high latencies, short timeouts, and other abnormal [Cluster metrics](/references/cluster-metrics).
+If the metrics come from a specific service (such as History Service), check that service's connectivity and restart if needed.
+
+#### Check Workflow logic
+
+Check your [Client and Worker configuration](/references/configuration) files for missing or invalid values in:
+
+- Server names
+- Network or host addresses
+- Certificates
+- Timeouts
+
+## Advanced troubleshooting
+
+If you're still getting this error, here are further things to check.
+
+#### After enabling mTLS
+
+Check the health of the Cluster with `tctl cluster health`.
+
+```
+tctl --address [SERVER_ADDRESS] cluster health
+```
+
+Add any missing [environment variables](/references/web-ui-environment-variables) to the configuration files, and correct any incorrect values.
+Server names and certificates must match between frontend and internode.
+
+#### After restarting the Temporal Cluster
+
+You might not be giving the Cluster enough time to respond and reconnect.
+Restart the Server, wait, and then check all services for connectivity and further errors.
+
+If the error persists, review your history and server logs for more specific causes before continuing to troubleshoot.
+
+#### When executing or scheduling Workflows
+
+One or more services may be unable to connect to the frontend.
+The Workflow might be unable to complete requests within the given connection time.
+
+Check service metrics, history logs, and connectivity between system components.
+If you're running the Cluster in a container, check that service's history or log files.
```

Comment:
This is repeated and can be removed. You've already mentioned checking service metrics, history logs, and connectivity.